### PR TITLE
19374 & 19375 amalgamation validaton fixes

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#19374 and /bcgov/entity#19375

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py [hunk 1]
```diff
@@ -181,15 +181,15 @@ def validate_amalgamating_businesses(  # pylint: disable=too-many-branches,too-m
...
+        (orgs := account_response.get('orgs')) and
+            any(str(org.get('id')) == account_id for org in orgs)):
         return True
     return False
 
 
 def _has_future_effective_filing(amalgamating_business: Business):
-    if Filing.get_filings_by_status(amalgamating_business.id,
-                                    [Filing.Status.PAID.value, Filing.Status.PENDING.value]):
+    if Filing.get_filings_by_status(amalgamating_business.id, [Filing.Status.PAID.value]):
```
Comment (argush3) [inline legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:192]: Why was "PENDING" status removed?  

Is the scenario where if only the invoice is created and the payment is not processed not considered pending filing?

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:192]: yes

**Review by argush3 — APPROVED** [review]

File: legal-api/src/legal_api/resources/v2/business/business.py [hunk 2]
```diff
@@ -90,8 +90,8 @@ def get_businesses(identifier: str):
                                 g.jwt_oidc_token_info,
                                 account_response)
         if orgs := account_response.get('orgs'):
-            if str(orgs[0].get('id')) == q_account:
-                business_json['accountId'] = orgs[0].get('id')
+            if any(str(org.get('id')) == q_account for org in orgs):
+                business_json['accountId'] = q_account
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/v2/business/business.py:94]: How about a code comment explaining why this is needed?

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py (hunk 1 above)
Comment (severinbeauvais) [inline legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:192]: I didn't check all the code in here, but is there also a check for "any filing not Completed or Corrected"? (These are the checks the UI does to ensure there is no pending filing.)

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py
Comment (severinbeauvais) [inline legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:1]: Is there already a check in here for 2 or more amalgamating businesses?

Re: /bcgov/entity#19359

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:1]: nope. do we have a backend ticket for that? i remember we discuss about validating duplicate amalgamating business. we can add this into that ticket?

Comment (severinbeauvais) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:1]: I will add this requirement to /bcgov/entity#19154.

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:192]: will change this logic based on the new requirement

File: legal-api/src/legal_api/resources/v2/business/business.py (hunk 2 above)
Comment (severinbeauvais) [reply legal-api/src/legal_api/resources/v2/business/business.py:97]: Looks good.

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Pull request merged**